lib/pulselistener.js

```diff
@@ -65,12 +83,21 @@ var PulseConnection = function(options) {
     namespace:          options.username || ''
   });
 
-  if (!options.connectionString) {
-    options.connectionString = buildPulseConnectionString(options);
-  } else {
+  this.isTaskclusterPulse = false;
+  if (options.connectionString) {
     assert(!options.username, "Can't take `username` along with `connectionString`");
     assert(!options.password, "Can't take `password` along with `connectionString`");
     assert(!options.hostname, "Can't take `hostname` along with `connectionString`");
+    assert(!options.contact, "Can't take `contact` along with `connectionString`");
+  } else if (options.username && options.password) {
+    // TODO: Deprecated warnings.
+    options.connectionString = buildPulseConnectionString(options);
+  } else {
+    assert(options.contact,
+      "Must provide `contact` when Pulse credentials are managed by taskcluster-pulse");
+    this._contact = options.contact;
+    this._pulse = new tc.Pulse();
```

Comment:
You don't need to keep reference to this object, I would create it when I need... especially as this code won't need it often...